firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FillAddressAutofillForm.swift

```diff
@@ -5,11 +5,38 @@
 import Foundation
 
 struct FillAddressAutofillForm: Codable {
-    let addressPayload: AddressAutofillPayload
-    let type: String
+    let payload: Payload
+    let type: FormAutofillPayloadType
 
-    enum CodingKeys: String, CodingKey, CaseIterable {
-        case addressPayload = "payload"
-        case type = "type"
+    struct Payload: Codable {
+        var addressLevel1: String
+        var addressLevel2: String
```

Comment:
Would be good to add a small test here for this codable as to make sure the payload we are expecting from formautofill is valid.

Why we need to test?
The payload can accidentally change when we import scripts or if someone accidentally pushes unwarranted changes.

Reply:
I agree. I think a better option would be to make them all optional since they're too impredictible. Also while we are here I noticed we have a discrepancy of naming the variables. In the telemetry callback we have ` "address_level1"` with underscore and here with hiphen  `"address-level1"`.
cc: @issammani

Reply:
Yes. That's one of the many annoying things I don't like about our data format. Once we move to glean we can fix this. Is it too much of a hassle to handle the different cases now in swift ? If that's the case, let's drop the extras until we move to glean on Desktop.

Reply:
Not at all, we just have to duplicate the decoding keys, that's all.

Reply:
Nice. Thanks. I will open a ticket from desktop side to match the names in the future.